### ISSUE TITLE
Revert shuffle page

### ISSUE THREAD

**Issue by mgursch** [Meetic/Shuffle#12]
Hi

need following feature:

User swipes a card away. 
User can restore last card by button click.

-> swipe back

Need a possibility to set adapter position (optional with animation to specific position)

Would be great if you can add this feature

**Comment by mgursch** [comment]
Are there some thoughts on this feature request?

**Comment by florent37** [comment]
Hi, you want a revert option ?

**Comment by mgursch** [comment]
Exactly

Just want an option to show the card before

eg
0,1,2,3,2,1,0

mShuffle.revert();
or 
mShuffle.setSelectedAdapterPosition(1, withAnimation);

**Comment by florent37** [comment]
the 2,1,0 will came from swipe or from a button outside of the shuffle ?

**Comment by mgursch** [comment]
in my specific case i need a button action to do this.

**Comment by florent37** [comment]
I'm on it ;)

**Comment by florent37** [comment]
can you try the 1.0.5 ? 👍 

```groovy
compile 'com.meetic.shuffle:shuffle:(last version)'
compile 'com.meetic.dragueur:dragueur:1.0.5'
```

[![](https://github.com/Meetic/Shuffle/raw/master/media/revert.gif)]

**Comment by mgursch** [comment]
Hey

thanks a lot for your support. I see an issue with that feature:

When I start the app it works for the first and second card. If i swipe further and do the revert one card is flickering.

Have tried it with images from web (using Glide) and images from local drawable resources. I uploaded a video here
https://goo.gl/photos/qPu47xWuU3XoVVxn8

Thanks a lot

**Comment by florent37** [comment]
I don't think it's an problem with the image with glide, I'll tried to investigate it

**Comment by mgursch** [comment]
The app  you see in the video is not using glide
its loading images via setImageResource - from local drawables folder. 

and there is no reason that glide hide an image when I call shuffle.revert. maybe a bug with the animation ?

**Comment by florent37** [comment]
I can reproduce it without using glide

**Comment by florent37** [comment]
Yes I think it's a bug with the animation

**Comment by mgursch** [comment]
Very good if you can reproduce it :) 

thanks again for your support.

**Comment by florent37** [comment]
I fixed it in the 1.0.6 👍
